### PR TITLE
Add basic static file building for ui_next

### PR DESCRIPTION
**Pull request by jakemcdermott**
For: #5957 

- [x] remove import aliases
- [x] add react-scripts
- [x] add proxy config for dev server
- [x] fix tests
- [x] update i18n commands and scripts
- [x] update docs

#### SUMMARY
This PR will be for adding basic static file building and other tooling for ui_next using `react-scripts`. You shouldn't need to change any of the build or development commands you're already using. `npm start`, `npm run test`, etc. should all just continue to work.

##### 1. Remove import aliases
`react-scripts` doesn't support webpack aliases. To avoid having to `eject` the default configuration, I removed our import aliases using [a python script](https://gist.github.com/jakemcdermott/95400da64e1088c0569053cb3d96775e). A log of the output can be found [here](https://gist.github.com/jakemcdermott/06140faf8db20312ded14db0a023faa8). When I'm done everything else, I might go back and try [an entirely different approach]( https://github.com/facebook/create-react-app/issues/2188#issuecomment-302040874) to at least preserve absolute imports.

##### 2. Add react-scripts
[react-scripts](https://github.com/facebook/create-react-app/tree/master/packages/react-scripts) is a commonly used set of build tools for react. It's easy to use, has very little configuration, and is well-supported. You may recognize it from `create-react-app`.

##### 3. add proxy config for dev server
We no longer have a webpack configuration to manage because react-scripts takes care of all that for us. That said, we still need a proxy configuration for the dev server. It can be found in [src/setupProxy.js](https://github.com/jakemcdermott/awx/blob/5957-statics/awx/ui_next/src/setupProxy.js). I tested this using the local dev environment and with an external awx backend.

##### 4. fix tests
- A few adjustments to the tests were needed. Most notably, components created with styled-components template literals need to have their display names defined explicitly to be selectable by enzyme now.

- The bootstrapping file for the jest test runner now lives in [src/setupTests.js](https://github.com/jakemcdermott/awx/blob/5957-statics/awx/ui_next/src/setupTests.js). This is where `react-script` expects to find this file and cannot be changed without `eject`ing the configuration.

##### 5. i18n
I moved the i18n config and generated source files from `build/locales` to `src/locales`. This is because `react-scripts` requires that all imported files live under `src/`. This actually makes sense and is the default location used by `lingui`

### CODE REVIEW

**Comment by jakemcdermott** [comment]
recheck

**Comment by jakemcdermott** [comment]
recheck

**Comment by keithjgrant** [comment]
It looks like the styled-components `css` prop isn’t working (we only use this in a handful of places I think). Here's the `VariablesDetail` component missing a few of its styles:

Screen Shot 2020-05-11 at 4 20 06 PM

(label isn't bold and the textarea isn't full width)

This can be fixed by adding this line to the top of any affected files:

```
import 'styled-components/macro';
```

([More info here](https://medium.com/styled-components/css-prop-support-for-create-react-app-37e8c5d96861))

**Comment by keithjgrant** [comment]
🎉 This appears to fix our broken HMR in Firefox 🎉

**Review by jlmitch5 — CHANGES_REQUESTED** [review]
this is great, it's cool that we don't have to worry about babel or webpack anymore.

looks like we probably need to update this dockerfile, it does a lot of add webpack.config stuff, and those files no longer exist.

https://github.com/ansible/awx/blob/e2fd11ebd1f2f8ee7d3d75a723146f5a996c1333/awx/ui_next/Dockerfile

File: awx/ui_next/babel.config.js [hunk 1]
```diff
@@ -1,18 +0,0 @@
-module.exports = api => {
-  api.cache(false);
-  return {
-    plugins: [
```
Comment (jlmitch5) [inline awx/ui_next/babel.config.js:4]: I guess we are now relying on whatever CRA deems as the babel stuff we can do.  I'm alright with that, provided they keep up with the types of new language features we want.  But do any of these not work anymore, and we need to update code to not include them?

File: awx/ui_next/package.json [hunk 2]
```diff
@@ -1,38 +1,66 @@
...
-    "@babel/plugin-proposal-class-properties": "^7.1.0",
-    "@babel/polyfill": "^7.0.0",
-    "@babel/preset-env": "^7.8.3",
-    "@babel/preset-react": "^7.0.0",
-    "@lingui/cli": "^2.7.4",
-    "@lingui/macro": "^2.7.2",
+    "@babel/polyfill": "^7.8.7",
+    "@lingui/cli": "^2.9.1",
+    "@lingui/macro": "^2.9.1",
     "@nteract/mockument": "^1.0.4",
```
Comment (jlmitch5) [inline awx/ui_next/package.json:62]: so it seems like we were including mockument before, but I couldn't figure out an equivalent to what is in the test setup file.  what exactly does it do (and does it do anythign different in this PR vs. the way it used to work?

File: awx/ui_next/package.json [hunk 3]
```diff
@@ -1,38 +1,66 @@
...
     "add-locale": "lingui add-locale",
     "extract-strings": "lingui extract",
     "compile-strings": "lingui compile",
     "prettier": "prettier --write \"src/**/*.{js,jsx,scss}\"",
     "prettier-check": "prettier --check \"src/**/*.{js,jsx,scss}\""
   },
-  "keywords": [],
-  "author": "",
-  "license": "Apache",
+  "browserslist": {
```
Comment (jlmitch5) [inline awx/ui_next/package.json:46]: this looks cool but I'm not entirely sure how it works, do you happen to know any details about this?

File: awx/ui_next/package.json (hunk 2 above)
Comment (jakemcdermott) [reply awx/ui_next/package.json:62]: mockument adds a polyfill for jsdom. if you comment this line out https://github.com/jakemcdermott/awx/blob/5957-statics/awx/ui_next/src/setupTests.js#L11, you'll see that many of our tests won't work without it. This is why we're using it on devel as well.

Comment (jlmitch5) [reply awx/ui_next/package.json:62]: sounds good, thanks for the explanation

File: awx/ui_next/package.json (hunk 3 above)
Comment (jakemcdermott) [reply awx/ui_next/package.json:46]: This helps react-scripts figure out what polyfills it is allowed to use by default. 

https://stackoverflow.com/a/56098322
https://create-react-app.dev/docs/supported-browsers-features/

File: awx/ui_next/babel.config.js (hunk 1 above)
Comment (jakemcdermott) [reply awx/ui_next/babel.config.js:4]: I think removing this file _did_  cause an issue (see https://github.com/ansible/awx/pull/6954#issuecomment-627017618) but with the latest commit I pushed, it should be fixed.

>  I'm alright with that, provided they keep up with the types of new language features we want.

Yep, exactly. Let's use it for now and if there ends up being some feature we absolutely have to add then we can weigh the pros and cons of not having the feature vs `ejecting` the configuration.

**Comment by jakemcdermott** [comment]
> looks like we probably need to update this dockerfile, it does a lot of add webpack.config stuff, and those files no longer exist.
> https://github.com/ansible/awx/blob/e2fd11ebd1f2f8ee7d3d75a723146f5a996c1333/awx/ui_next/Dockerfile

@jlmitch5 Damn good find - thanks for catching that! I'll fix that up now.

**Comment by jakemcdermott** [comment]
> It looks like the styled-components `css` prop isn’t working (we only use this in a handful of places I think). Here's the `VariablesDetail` component missing a few of its styles:
> (label isn't bold and the textarea isn't full width)
> 
> This can be fixed by adding this line to the top of any affected files:
> 
> ```
> import 'styled-components/macro';
> ```
> 
> ([More info here](https://medium.com/styled-components/css-prop-support-for-create-react-app-37e8c5d96861))

@keithjgrant thanks for catching that - should be fixed now

File: awx/ui_next/babel.config.js (hunk 1 above)
Comment (jlmitch5) [reply awx/ui_next/babel.config.js:4]: makes sense to me.  my guess is it won't be a problem, they'll probably be ahead of the curve on that stuff

Comment (jlmitch5) [reply awx/ui_next/babel.config.js:4]: ah actually, it looks like the babel macros (like the styled components import to get css to work) were created for the express purpose of getting new language features without having to eject from CRA/other zero-config build systems.  So if this situation pops up, we can just look if there is a macro for whatever we want to add.

https://babeljs.io/blog/2017/09/11/zero-config-with-babel-macros

**Comment by jakemcdermott** [comment]
@jlmitch5 @keithjgrant I believe this is ready for another look

**Review by keithjgrant — APPROVED** [review]
🎉

**Review by jlmitch5 — APPROVED** [review]
looks good

**Review by marshmalien — APPROVED** [review]
Pulled down and did a bit of manual testing. I didn't find any ~glaring~ issues 🎉

**Comment by nixocio** [comment]
Same here. Pulled down locally and everything seems to be working fine. Good job @jakemcdermott .

**Review by nixocio — APPROVED** [review]

**Pull request merged**